docs/cli/graphql-transformer/directives.md

```diff
@@ -2327,4 +2327,227 @@ Each of the actions described in the @predictions definition section can be used
+Host: www.example.com
+```
+
+**Request Body**
+
+The `@http` directive also allows you to specify the body of a request, which is used for `POST`, `PUT`, and `PATCH` requests. To create a new post, you can define the following.
+
+
+```graphql
+type Query {
+  addPost(title: String!, description: String!, views: Int): Post @http(
+    method: POST
+    url: "https://www.example.com/post"
+    )
+}
+```
+
+Then add a post by using the body input type:
+
+```graphql
+mutation add {
+  addPost(body: {title: "new post", description: "fresh content"}) {
+    id
+  }
+}
+```
+
+**Specifying the environment**
+
+The `@http` directive allows you to use `${env}` to reference the current Amplify CLI environment.
+
+```graphql
+type Query {
+  listPosts: Post @http(
+    url: "https://www.example.com/${env}/posts"
+  )
+}
+```
+
+**Combining the different components**
+
+You can use a combination of params, query, body, headers, and environments in your `@http` directive definition.
+
+Given the definition
+
+```graphql
+type Query {
+  updatePost(title: String!, description: String!, views: Int, withComments: Boolean): Post @http(
+    method: PUT
+    url: "https://www.example.com/${env}/posts/:id"
+    headers: [{key: "X-Header", value: "X-Header-Value"}]
+    )
+}
+```
+
+you can update a post with
+
+```graphql
+mutation update {
+  updatePost(
+    body: {title: "new title", description: "updated description", views: 100}
+    params: {id: "EXISTING_ID"}
+    query: {withComments: true}) {
+    id
+    title
+    description
+    comments {
+      id
+    }
+  }
+}
+```
+
+which will send
+
+```
+PUT /ENV/posts/EXISTING_ID?withComments=true
+Host: www.example.com
+X-Header: X-Header-Value
+{
+  title: "new title"
+  description: "updated description"
+  views: 100
+}
+```
+
+**Advanced cases**
+
+In some cases, you may want to send a request based on existing field data. Take a scenario where you have a post and want to fetch comments associated with the post in a single query. You can fetch the comments based on the post id with the following definition. `$ctx.source` is a map that contains the resolution of the parent field (`Post`) and gives access to `id`.
+
+
+```graphql
+type Post {
+  id: ID!
+  comments: [Comment] @http(
+    url: "https://www.example.com/posts/${ctx.source.id}/comments"
+    )
+}
+```
+
+You can retrieve the comments of a specific post with the following query and selection set.
+
+```graphql
+query post {
+  getPost(params: {id: "POST_ID"}) {
+    id
+    comments {
+      id
+      content
+    }
+  }
+}
+```
+
+Assuming that getPost retrieves a post with the id `POST_ID`, the comments field is resolved by sending this request to the endpoint
+
+```
+GET /posts/POST_ID/comments
+Host: www.example.com
+```
+
+Note that there is no check to ensure that the reference variable (here the post ID) exists. When using this technique, it is recommended to make sure the referenced field is non-null.
+
+### Generates
```

Comment:
We should note that the current implementation of the @http transformer does not support the calling of other AWS services where SIG4 signature is required.